### PR TITLE
LPD-24498 Notification template for an order

### PR DESCRIPTION
**Pull request by smottal**
https://liferay.atlassian.net/browse/LPD-24498

Resent from: https://github.com/liferay-commerce/liferay-portal/pull/5066 not using batch engine

### CODE REVIEW

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:report:24489861

**Comment by smottal** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/notification/term/evaluator/CommerceOrderItemsNotificationTermEvaluator.java [hunk 1]
```diff
@@ -148,18 +159,32 @@ private Map<String, Object> _getOrderItem(
...
+					return portalURL +
+						_commerceMediaResolver.getDefaultURL(
+							cpDefinition.getGroupId());
+				}
 
-				String cpInstanceThumbnailSrc =
-					_cpInstanceHelper.getCPInstanceThumbnailSrc(
-						AccountConstants.ACCOUNT_ENTRY_ID_ADMIN,
-						commerceOrderItem.getCPInstanceId());
+				String imageURL = _commerceMediaResolver.getURL(
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/notification/term/evaluator/CommerceOrderItemsNotificationTermEvaluator.java:183]: Nice!

**Comment by smottal** [comment]
@luca-pellizzon if you can have a look to the updated classes: the test and the PortalInstanceLifecycleListener - I used this instead of the VerifyProcess because it's not enough (it doesn't cover the creation of a new company for example) and with @riccardo-alberti we moved on the lifecycle

File: modules/apps/commerce/commerce-notification-service/src/main/java/com/liferay/commerce/notification/internal/instance/lifecycle/AddCommerceOrderNotificationPortalInstanceLifecycleListener.java [hunk 2]
```diff
@@ -196,7 +226,32 @@ private void _verifyCommerceOrderObjectAction(long companyId)
 	@Reference
 	private RoleLocalService _roleLocalService;
 
+	private ServiceRegistration<FeatureFlagListener> _serviceRegistration;
+
 	@Reference
 	private UserLocalService _userLocalService;
 
+	private class FeatureFlagListenerImpl implements FeatureFlagListener {
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-notification-service/src/main/java/com/liferay/commerce/notification/internal/instance/lifecycle/AddCommerceOrderNotificationPortalInstanceLifecycleListener.java:234]: you've already created a feature flag listener, what is this for?

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:report:25054076

**Comment by smottal** [comment]
ci:forward:force

**Pull request closed without merging**